add-item completed

rishnarenikunta opened the pull request:
Completed the add item page, with ability to upload picture, store the item name, count, and category chosen. I attached a picture with the mobile version and desktop. 
<img width="541" alt="Screenshot 2024-10-30 at 11 04 52 PM" src="https://github.com/user-attachments/assets/e6618b0f-8e78-47b7-b398-3e732d445957">
<img width="302" alt="Screenshot 2024-10-30 at 11 05 27 PM" src="https://github.com/user-attachments/assets/e10a210c-17ed-4963-bdfc-5780a0822dd7">

emoryjgrubbs commented:
make the tag a div, and slot the module into the div that says "add items" in inventory management, it should already be set up to open on clicking the plus (you will need to make it absolute position so it will float above other content)

you should avoid mixing tailwind with a lot of css

switch to a a hero icon x so it's more square

the add item header text should probably also be a heavier weight

reduce the width of the submit on mobile a bit, and probably make it a bit wider on desktop

CrazEpic commented on components/AddItem.vue:
```diff
@@ -0,0 +1,131 @@
+  isDropdownOpen.value = false;
+}
+
+function triggerFileInput() {
```
instead of using query selectors, you can wrap the input tag under a label so that you can do the styling you need

in fact, it's good practice to have a label element and a corresponding input element 
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label

example
```pug
label(for="fileInput")
    some box
        the other box saying "Upload File"
    input(id="fileInput")
```

CrazEpic commented on components/AddItem.vue:
```diff
@@ -0,0 +1,131 @@
+  const target = event.target as HTMLInputElement;
+  if (target.files && target.files.length > 0) {
+    imageFile.value = target.files[0];
+    console.log('Uploaded file:', imageFile.value);
```
good for debugging, but don't put leave console logs in a PR

CrazEpic commented on components/AddItem.vue:
```diff
@@ -0,0 +1,131 @@
+</template>
+
+<script lang="ts" setup>
+import { ref, watch } from 'vue';
```
imports not needed, already auto imported

CrazEpic commented on components/AddItem.vue:
```diff
@@ -0,0 +1,131 @@
+    div.flex.justify-between.items-center.w-full.mb-4.bg-utd-orange.rounded-t-lg.p-1
+      h2.text-lg.pl-4.text-white.mx-auto.font-semibold Add Item
+      button.text-lg.text-white.pr-2(@click="closeForm")
+        svg(xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="white" stroke="white" class="size-6")
```
don't hardcode the icons with svg, we have https://heroicons.com/

CrazEpic commented on components/AddItem.vue:
```diff
@@ -0,0 +1,131 @@
+        button.bg-utd-green.text-white.h-10.px-5.py-2.rounded-md.flex.items-center.gap-5.cursor-pointer.shadow-md.font-semibold(@click="triggerFileInput"
+          :class="{'bg-gray-500 bg-opacity-60 text-white': imageUrl}"
+          style="position: relative; z-index: 1;")
+          svg(xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2.0" stroke="white" class="size-6")
```
don't hardcode the icons with svg, we have https://heroicons.com/

CrazEpic commented on components/AddItem.vue:
```diff
@@ -0,0 +1,131 @@
+      div.relative.flex.flex-col.items-center.justify-center.max-w-48.max-h-48.min-w-48.min-h-48.rounded.cursor-pointer.bg-gray-300.shadow-lg.mt-16.mr-10.mb-12.margin-adjust
+        img(v-if="imageUrl" :src="imageUrl" class="absolute inset-0 w-full h-full object-cover rounded shadow-lg")
+        button.bg-utd-green.text-white.h-10.px-5.py-2.rounded-md.flex.items-center.gap-5.cursor-pointer.shadow-md.font-semibold(@click="triggerFileInput"
+          :class="{'bg-gray-500 bg-opacity-60 text-white': imageUrl}"
```
why are we making class reactive?

rishnarenikunta replied:
bc the style of the upload button changes if there is a imageURL in place, thus making it reactive

CrazEpic commented on components/AddItem.vue:
```diff
@@ -0,0 +1,131 @@
+   div.flex.flex-col.items-center.rounded-lg.drop-shadow-standard.m-3.mobile-layout.bg-white
+    div.flex.justify-between.items-center.w-full.mb-4.bg-utd-orange.rounded-t-lg.p-1
+      h2.text-lg.pl-4.text-white.mx-auto.font-semibold Add Item
+      button.text-lg.text-white.pr-2(@click="closeForm")
```
this button does nothing lol, the close button doesn't work

CrazEpic commented on components/AddItem.vue:
```diff
@@ -0,0 +1,131 @@
+          | Upload
+          input(type="file" ref="fileInput" style="display: none" @change="handleFileUpload")
+      div.flex.flex-col
+        div.w-full.flex.flex-col.mb-5
```
no need to wrap the input within a div, just style the input element directly

CrazEpic commented on components/AddItem.vue:
```diff
@@ -0,0 +1,131 @@
+          button.w-full.bg-utd-orange.text-white.py-2.rounded-xl.shadow-md.font-semibold(@click="isDropdownOpen = !isDropdownOpen") {{ selectedCategory || 'Category' }}
+          div(v-if="isDropdownOpen" class="absolute w-full bg-white text-black border border-gray-300 rounded-xl shadow-md z-10 overflow-hidden")
+            ul.max-h-40.overflow-y-auto.scroll-visible.text-center
+              li(v-for="filter in categories" :key="filter.name"
```
use proper naming, there is no "filtered category" when selecting a category for add item

CrazEpic commented on components/AddItem.vue:
```diff
@@ -0,0 +1,131 @@
+                | {{ filter.name }}
+              
+    div.submit-footer.rounded-xl
+      button.submit-button.w-full.h-10.px-14.bg-utd-green.text-white.rounded-xl.shadow-md.font-semibold(@click="submitForm") Submit
```
no function?

rishnarenikunta commented on public/test-images/Grains/pasta.png:
was causing issues, deleted it while working with Emory

rishnarenikunta commented on public/test-images/Grains/ramen.png:
was causing issues, deleted it while working with Emory

rishnarenikunta commented on pages/inventory-management.vue:
```diff
@@ -19,7 +19,8 @@ div
                 | searchterm = {{searchTerm}}
             div.flex.flex-row.items-center.justify-center
-        p( v-if="addItemsOpen" ).text-center.pt-20 Add Items
+        div( v-if="addItemsOpen" ).text-center.pt-20.flex.justify-center.items-center.absolute.inset-0
```
I tried doing that but it is not positioning the component correctly. I will leave it like this for now if thats okay.

CrazEpic commented on pages/inventory-management.vue:
oh, and that is not how you pass a prop

CrazEpic commented on components/AddItem.vue:
```diff
@@ -0,0 +1,77 @@
+                | {{ category.name }}
+
+    div.sticky.bottom-0.w-full.flex.justify-center
+      button.mb-5.h-10.px-14.bg-utd-green.text-white.rounded-xl.shadow-md.font-semibold.sm_w-auto(class="w-4/6") Submit
```
responsive width styling is wrong here, with cascade in CSS, between two classes the class defined later has higher precedence when styling

thus your width is always w-4/6

rishnarenikunta commented:
I fixed the tabbing, added a drop down icon, and fixed some of the styling. However, there is still one issue that I have been attempting to fix for the last couple hours. The add item modal does not cover the entirety of the screen for mobile (does not cover the header as well). I attempted to fix it with things like z-index, but it is not working. I attached a picture of how it looks currently (just implemented some padding for now). I can revisit this issue again as I think it would be best to move on so I can get some progress on more important tasks. 

<img width="378" alt="Screenshot 2024-11-14 at 11 17 35 PM" src="https://github.com/user-attachments/assets/e877cb56-5f23-42b4-a93f-bd42526c98a5">

CrazEpic commented on components/AddItem.vue:
```diff
@@ -0,0 +1,72 @@
+					input(type="file" id="fileInput" style="display: none" @change="handleFileUpload")
+			div.flex.flex-col
+				input.w-full.mb-5.text-md.border-b-2.border-black.p-1(v-model="itemName" type="text" placeholder="Item Name")
+				input.w-full.mb-5.text-md.border-b-2.border-black.p-1(v-model="itemCount" type="number" placeholder="Item Count")
```
add min="0"
